feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/utils.c

```diff
@@ -222,6 +222,67 @@ int phy_get_channels(const char *name, int *channel)
 	return j;
 }
 
+int phy_get_channels_state(const char *name, struct schema_Wifi_Radio_State *rstate)
+{
+	struct wifi_phy *phy = phy_find(name);
+	int i, j = 0, k = 0, l = 0;
+	int dis[64], allow[64], radar[64];
+	int index = 0, ret = 0;
+	char str[256];
+	
+	if (!phy)
+		return 0;
+
+	for (i = 0; (i < IEEE80211_CHAN_MAX); i++) {
+			if (phy->chandisabled[i]) {
+				dis[j] = i;
+				j++;
+			}
+			else if (phy->chandfs[i]) {
+				radar[k] = i;
+				k++;
+
+			}			
+			else if (phy->channel[i]){
+				allow[l] = i;
+				l++;
+			}
+		}
+
+
+		if(j){
+			STRSCPY(rstate->channels_keys[ret], "disabled");
+			for (i=0; i<j; i++){
+   				index += sprintf(&str[index], "%d,", dis[i]);
+			}
+				memcpy(rstate->channels[ret], str, index);
+				index = 0;
+				ret++;
+		}
+
+		if(k){
+			STRSCPY(rstate->channels_keys[ret], "radar_detection");
+			for (i=0; i<k; i++){
+   				index += sprintf(&str[index], "%d,", radar[i]);
+			}
+				memcpy(rstate->channels[ret], str, index);
+				index = 0;
+				ret++;
+		}
+
+		if(l){
+			STRSCPY(rstate->channels_keys[ret], "allowed");
+			for (i=0; i<l; i++){
+   				index += sprintf(&str[index], "%d,", allow[i]);
+				}
+				memcpy(rstate->channels[ret], str, index);
+				index = 0;
+				ret++;
+		}
```

Comment:
Just a thought. May be you can add a new function to populate the "channels" map. This would increase readability and you can get rid of some duplicate code. Something like this 

static void update_channels_state(struct schema_Wifi_Radio_State *rstate,
				      int *index, const char *key, const char *value)
{
	STRSCPY(rstate->channels_keys[*index], key);
	STRSCPY(rstate->channels[*index], value);

	*index = *index + 1;
	rstate->channels.len = *index;
}

And then simply call this above
if(j)
     update_channels_state(rstate, &index, "disabled", dis);
Similarily for other keys
Disclaimer: Code is not tested